lib/unit_test/helper.js

```
@@ -230,65 +236,109 @@ function outputJUnitXML(jsonResults, prefix, next) {
  * @param  {Function} callback  [description]
  */
 function test(sdkSrc, platforms, callback) {
-	var sdkPath,
-		tasks = [],
+	let sdkPath,
 		results = {};
 
-	tasks.push(function (next) {
-		// install new SDK and delete old test app in parallel
-		async.parallel([
-			function (cb) {
-				if (sdkSrc) {
-					console.log('Installing SDK');
-					AppcUtil.installSDK(sdkSrc, cb);
-				} else {
-					cb();
-				}
-			},
-			clearPreviousApp
-		], next);
-	});
-	// Record the SDK we just installed so we retain it when we clean up at end
-	tasks.push(function (next) {
-		AppcUtil.getSDKInstallDir(function (err, installPath) {
-			if (err) {
-				return next(err);
-			}
-			sdkPath = installPath;
-			next();
+	Promise.resolve()
+		.then(() => {
+			// install new SDK and delete old test app in parallel
+			return new Promise(resolve => {
+				initEnvironment(sdkSrc, resolve);
+			});
+		})
+		.then(() => {
+			// Record the SDK we just installed so we retain it when we clean up at end
+			return new Promise((resolve, reject) => {
+				AppcUtil.getSDKInstallDir(function(err, installPath) {
+					if (err) {
+						return reject(err);
+					}
+					sdkPath = installPath;
+					resolve();
+				});
+			});
+		})
+		.then(() => {
+			return new Promise(resolve => {
+				console.log('Generating project');
+				AppcUtil.generateProject(PROJECT_NAME, __dirname, platforms, resolve);
+			});
+		})
+		.then(() => {
+			return new Promise(resolve => {
+				copyMochaAssets(resolve);
+			});
+		})
+		.then(() => {
+			return new Promise(resolve => {
+				addTiAppProperties(resolve);
+			});
+		})
+		.then(() => {
+			return new Promise(resolve => {
+				build(platforms, results, resolve);
+			});
+		})
+		.then(() => {
+			AppcUtil.cleanNonGaSDKs(sdkPath, function(cleanupErr) {
+				callback(cleanupErr, results);
+			});
+		})
+		.catch((error) => {
+			console.log(error);
```

We need to replace this line with `callback(error);` so, we can bubble up any errors back to `test.js`.